hi-ml-azure/tests/testazure/test_azure_util.py

```diff
@@ -452,15 +513,31 @@ def test_determine_run_id_source(tmp_path: Path) -> None:
     mock_args = parser.parse_args(["--experiment", "fake_experiment"])
     assert util.determine_run_id_source(mock_args) == util.AzureRunIdSource.EXPERIMENT_LATEST
 
-    # If run recovery id is provided, expect source to be that
-    mock_args = parser.parse_args(["--run_recovery_ids", "experiment:run1234"])
+    parser = ArgumentParser()
+    parser.add_argument("--run_recovery_id", type=str)
+    parser.add_argument("--run_id", type=str)
```

Comment:
This is not an ideal pattern for writing tests. You are defining a parser here, and check if determine_run_id_source still does the right thing. But your actual code uses a parser defined in the download script - if you accidentally made a change to that parser, your test here would still pass, and hence not provide much value. It would be better if you refactored the parser creation in the download script such that the parser creation lives in a function, that you can invoke here in the test to create a parser.

Reply:
This parser doesn't replicate any parser from any one script - the idea was to add all possible run sources to it -hence why it has both run_id and run_ids which no single script would have